sklearn/manifold/t_sne.py

```diff
@@ -142,10 +142,12 @@ def _kl_divergence(params, P, degrees_of_freedom, n_samples, n_components,
 
     # Q is a heavy-tailed distribution: Student's t-distribution
     n = pdist(X_embedded, "sqeuclidean")
-    n += 1.
```

Comment:
Sorry for the slow reply @jnothman !
The test case works by testing for degrees_of_freedom > 1, "alpha" in the test case refers to the number of degrees of freedom. The reason for this is because the calculation of KL divergence doesn't depend on n_components (that just changes the dimensions of the matrix)
However, I could also create a test case for n_components > 2 if you like?